vm/src/array.rs

```diff
@@ -7,12 +7,17 @@ use std::slice;
 
 use gc::{Gc, Traverseable};
 
+mod internal {
+    pub struct CantConstruct(());
+}
+
 /// Abstract array type which have their length store inline with the elements.
 /// Fills the same role as Box<[T]> but takes only 8 bytes on the stack instead of 16
 #[repr(C)]
 pub struct Array<T: Copy> {
     len: usize,
     array_start: [T; 0],
+    _marker: self::internal::CantConstruct,
```

Comment:
I thought just having public fields meant that folks couldn't construct it 🤔

Reply:
I think the intention is that it shouldn't be constructed even accidentally from *this module*.